addons/stock/models/product.py

```diff
@@ -631,6 +635,29 @@ def write(self, vals):
     def action_open_quants(self):
         return self.product_variant_ids.action_open_quants()
 
+    def action_update_quantity_on_hand(self):
+        if (self.env.user.user_has_groups('stock.group_stock_multi_locations') or
+        self.env.user.user_has_groups('stock.group_production_lot') or
+        self.env.user.user_has_groups('stock.group_tracking_owner') or
+        self.env.user.user_has_groups('product.group_stock_packaging')):
+            return self.action_open_quants()
+        else:
+            default_product_id = self.env.context.get('default_product_id', self.product_variant_id.id)
+            wiz = self.env['stock.change.product.qty'].create({
```

Comment:
are you sure you need to create the wizard and return it instead of returning an action and the values in default_ context keys?